frontend/src/store/clientSettings.ts

```diff
@@ -68,107 +81,93 @@ class ClientSettingsStore {
-          storeKey,
-          state.value
-        );
-
-        if (data) {
-          syncDataWatcher.pause();
-          Object.assign(state.value, data);
-          await nextTick();
-          syncDataWatcher.resume();
-        }
-      } catch {
-        const { t } = usei18n();
-
-        useSnackbar(t('failedSettingDisplayPreferences'), 'error');
+  public constructor() {
+    const remote = useRemote();
+    /**
+     * == WATCHERS ==
+     */
+
+    /**
+     * Sync data with server
+     */
+    const syncDataWatcher = watchPausable(state, async () => {
+      if (remote.auth.currentUser) {
+        await preferencesSync(storeKey, state.value);
       }
-    }
-  },
-  { immediate: true }
-);
-
-/**
- * Sync data with server
- */
-const syncDataWatcher = watchPausable(
-  state,
-  async () => {
-    if (remote.auth.currentUser) {
-      await preferencesSync(storeKey, state.value);
-    }
-  },
-  { deep: true }
-);
-
-/**
- * Locale change
- */
-const BROWSER_LANGUAGE = computed<string>(() => {
-  const rawString = useNavigatorLanguage().language.value || '';
-  /**
-   * Removes the culture info from the language string, so 'es-ES' is recognised as 'es'
-   */
-  const cleanString = rawString.split('-');
-
-  return cleanString[0];
-});
-
-watch(BROWSER_LANGUAGE, () => (clientSettings.locale = BROWSER_LANGUAGE.value));
-
-watch(
-  () => state.value.locale,
-  () => {
-    const i18n = usei18n();
+    });
 
-    i18n.locale.value =
-      state.value.locale !== 'auto'
-        ? state.value.locale
-        : BROWSER_LANGUAGE.value || String(i18n.fallbackLocale.value);
-  },
-  { immediate: true }
-);
+    /**
+     * Fetch data when the user logs in
+     */
+    watch(
+      () => remote.auth.currentUser,
+      async () => {
+        if (remote.auth.currentUser) {
+          try {
+            const data = await fetchSettingsFromServer<ClientSettingsState>(
+              storeKey,
+              state.value
+            );
+
+            if (data) {
+              syncDataWatcher.pause();
+              Object.assign(state.value, data);
+              await nextTick();
+              syncDataWatcher.resume();
+            }
+          } catch {
+            const { t } = usei18n();
+
+            useSnackbar(t('failedSettingDisplayPreferences'), 'error');
+          }
+        }
+      },
+      { immediate: true }
+    );
+
+    /**
+     * Locale change
+     */
+
+    watch(BROWSER_LANGUAGE, () => (this.locale = BROWSER_LANGUAGE.value));
+
+    watch(
+      () => this.locale,
+      () => {
+        const i18n = usei18n();
+
+        i18n.locale.value =
+          this.locale !== 'auto'
+            ? this.locale
+            : BROWSER_LANGUAGE.value || String(i18n.fallbackLocale.value);
+      },
+      { immediate: true }
+    );
+
+    /**
+     * Vuetify theme change
+     */
+    watch(browserPrefersDark, () => {
+      state.value.darkMode = browserPrefersDark.value;
+    });
```

Comment:
@ThibaultNocchi I don't agree with this, many apps (at least on phones, but it's true not all of them do because they're not ready to swap themes while running) react to the system color change.

Perhaps we should add an additional `auto` state once we move the button to the user menu and settings. Either way, I expect the any change I perform to the color scheme to stick. Reverting it back its an easy click anyway.

Reply:
FWIW, I agree that your OS automatically changing it's theme and having that update in the UI is cool and reasonably expected. I don't think it should overwrite a set preference by a user unless it's been set to an `auto` mode.

Reply:
@aweebs @ThibaultNocchi Auto mode implemented. Closing this.